.gitlab-ci.yml

```diff
@@ -0,0 +1,87 @@
+# .gitlab-ci.yml
+# rust-secp256k1
+
+
+stages:
+  - test
+  - build
+
+
+
+image:                             parity/rust-builder:latest
+
+variables:
+  GIT_STRATEGY:                    fetch
+  CARGO_HOME:                      "/ci-cache/${CI_PROJECT_NAME}/cargo/${CI_JOB_NAME}"
+  SCCACHE_DIR:                     "/ci-cache/${CI_PROJECT_NAME}/sccache"
+  CI_SERVER_NAME:                  "GitLab CI"
+  DOCKER_OS:                       "debian:stretch"
+  ARCH:                            "x86_64"
+
+
+.collect-artifacts:                &collect-artifacts
+  artifacts:
+    name:                          "${CI_JOB_NAME}_${CI_COMMIT_REF_NAME}"
+    when:                          on_success
+    expire_in:                     7 days
+    paths:
+      - artifacts/
+
+.docker-env:                       &docker-env
+  tags:
+    - linux-docker
+
+.compiler_info:                    &compiler_info
+  before_script:
+    - rustup show
+    - cargo --version
+    - sccache -s
+
+.build-refs:                       &build-refs
+  only:
+    - master
+    - schedules
+    - web
+    - /^v[0-9]+\.[0-9]+.*$/        # i.e. v1.0, v2.1rc1
+
+.test-refs:                        &test-refs
+  only:
+    - master
+    - schedules
+    - web
+    - /^v[0-9]+\.[0-9]+.*$/        # i.e. v1.0, v2.1rc1
+    - /^[0-9]+$/
+
+
+
+
+
+test-linux-stable:                 &test
+  stage:                           test
+  <<:                              *test-refs
+  <<:                              *docker-env
+  <<:                              *compiler_info
+  variables:
+    RUST_TOOLCHAIN: stable
+    RUSTFLAGS: -Cdebug-assertions=y
+    TARGET: native
+  script:
+    - time cargo test --all --release --verbose
+    - sccache -s
+
+
+
+
+build-linux-release:               &build
+  stage:                           build
+  <<:                              *build-refs
+  <<:                              *docker-env
+  <<:                              *compiler_info
+  <<:                              *collect-artifacts
+  script:
+    - time cargo build --release --verbose
+    - mv ./target/release/libsecp256k1* ./artifacts/.
```

Comment:
Are these artifacts going to be reused by parity-ethereum? Otherwise there is no point in collecting them.